app/src/main/java/ch/epfl/skysync/screens/admin/AddUser.kt

```diff
@@ -116,51 +116,28 @@ fun AddUserScreen(navController: NavController) {
                 emailError = !validateEmail(emailValue)
                 firstNameError = textInputValidation(firstNameValue)
                 lastNameError = textInputValidation(lastNameValue)
-                roleError = textInputValidation(roleValue)
+                roleError = !inputNonNullValidation(roleValue)
```

Comment:
Here you could create a function/lambda (for ex: `createUser`) to have a slightly cleaner code

Reply:
Well I don't know if it will be really better as it will require a function with a lot of parameters

Reply:
Right I was thinking of something like
```kotlin
val createUser: () -> Unit = {
    // create user
}
```
that you would put in `AddUserScreen`, but I don't know if it's Kotlin proof (it is more a JS-like approach)